### DISCUSSION TITLE
WithWorkdir doesn't work as expected in existing dagger session

### DISCUSSION THREAD

**Discussion by sipsma** [dagger/dagger#12485] [Help]
If you connect to an existing dagger session (e.g. via dagger-in-dagger or via `dagger run`) and specify a workdir it's ignored and instead the workdir will be the workdir of the session process.

I think the very minimum would be to throw an error if this happens (a workdir other than the one of the session is specified) just to at least replace the confusing behavior w/ annoying but comprehensible behavior.

It also might be possible to actually allow different clients to have different workdir settings. It would require the workdir get plumbed in each request (an http header?) through to every part of the session code where workdir is utilized. There might be some gotcha lurking but at first thought that feels possible.

**Comment by shykes** [comment]
I would expect that multiple clients sharing the same session would never happen, unless for advanced debugging reasons. Certainly I would *not* expect `dagger run` or dagger-in-dagger to reuse a session - I would expect each client to get a fresh session each time. I'm guessing that's not how it works today (hence this issue), but can we talk about the constraints around that, and whether we could change them?

**Comment by helderco** [comment]
This has been fixed in the Go SDK:
- https://github.com/dagger/dagger/pull/5126

Need the same for the others, but it's not just workdir. If a session is already opened, there's a few config settings that don't apply. Any of those could raise an error if set, and at least adjust their docstring so the user doesn't get surprised by the ignored behavior.

However, since today we can run a script independently with `python main.py`, or as part of `dagger run python main.py`, the user would have to make changes to the code when switching so 🤷.

**Comment by TomChv** [comment]
This has been fix on Node too (#5675)
